### DISCUSSION TITLE
Address book alternatives for free users

### DISCUSSION THREAD

**Discussion by RaXorX** [rustdesk/rustdesk#6161] [General]
@rustdesk 

I had opened a q&a thread a couple days ago hoping to have someone answer the queries I had. https://github.com/rustdesk/rustdesk/discussions/6089

Rustdesk being an oss is already doing a great job than teamviewer and anydesk so far. But as much as I understand the need for sourcing funds for the continued development, I fear forcing users to pay for monthly subscriptions for features they do not particularly need/require is a bit of a corporate-tier monopoly and very unseemly of an oss. 

In the thread I specified how the packages are tailored for company usage rather than home usage, and thus being charged the same for a few basic features as a home user is a bit too excessive. Yet I have still to hear from @rustdesk or any other moderator on that behalf. 

Since the question thread is left unanswered, I'm opening a separate feature request thread here. TeamViewer used to be pretty good aside from its adware. Even Anydesk allows a minimum of 5 address book and local backup/restore option for it (or they used to). Aside from the time outs now they have, rustdesk is simply following in the footsteps of Anydesk.
Since rustdesk is OSS I'd simply like to propose some alternatives to address book feature for free users. 

A) *Provide a separate cheaper package for home users. 10$/mo just to have managed devices and address book seems too excessive. Pricing is up to you but 1$/mo seems fair as a starting price. (More than 2$ would be too excessive for those features)

B)
    i) Provide an option to have local address book. Since it is local, I don't see the need to put a limit to this.
    ii) Provide option to backup and restore address book.
    iii) Provide an option to sync address book to a host of our choice. (this can either be provided as periodic sync or manual or both)
    iv) Provide an option to sync address book to cloud.
ii, iii, iv can also be extended to settings, in fact there should be an option to export and import settings if it already doesn't exist.

*This whole private hosting subscription seems like a good option for private hosting for companies who want their servers and data to be separate from everyone else if they can't host their own on their own. Rustdesk giving that option to host one on their behalf is pretty good. But that's all it is at best. Any home user would never require or have a need for private hosting. 
I asked in the q&a thread whether the servers log IPs or not. And if they are already doing that, how much difference would it really make storing a couple extra bytes of data which would consist of the user accounts, address book and keeping a list of managed devices on your servers?

It would make sense if you're not storing any data on the users for privacy in which case logging IPs would defeat that purpose. It would also make sense if you didn't want to store private data of users on your public servers (central/relay). It would also make sense if the hosting was done on 3rd party contracted servers and not your own data servers. But since you guys are already relaying the streams for free for which you have to pay from your ends, I don't see the reason why you can't also provide data storage for user access, accounts and address book for free users. Maybe a little limited perhaps but not too limited. 

Otherwise it hardly seems any reasonable at all keeping those features locked up behind a paywall suitable for corporate users. Unless this software is being tailored or made for corporate users more. And still, after all this you cannot provide these features (managed devices, address book) to free users, then please consider the alternatives I mentioned.

Make a difference where you can. Just being an OSS doesn't make it any different or better. As most OSS I've usually tend to be tailored for corporate users. I sincerely hope rustdesk is not one of them.

**Comment by RaXorX** [comment]
TLDR:

Since rustdesk is OSS I'd simply like to propose some alternatives to address book feature for free users.

A) *Provide a separate cheaper package for home users. 10$/mo just to have managed devices and address book seems too excessive. Pricing is up to you but 1$/mo seems fair as a starting price. (More than 2$ would be too excessive for those features)

B)
i) Provide an option to have local address book. Since it is local, I don't see the need to put a limit to this.
ii) Provide option to backup and restore address book.
iii) Provide an option to sync address book to a host of our choice. (this can either be provided as periodic sync or manual or both)
iv) Provide an option to sync address book to cloud.
ii, iii, iv can also be extended to settings, in fact there should be an option to export and import settings if it already doesn't exist.

**Reply by rivman** [reply]
I would gladly pay 1-2/month for adress book. Now I use a simple google sheet in my phone. Take me 2 seconds to find what I look for. But still, I want to use RustDesk directly.

**Comment by lazee486** [comment]
So far I've noticed the rustdesk:// handler so looking at a shared excel or Google sheets document with links... or integrate it with tactical rmm...

**Comment by kal10bach** [comment]
try remote desktop manager from devolutions,net, there's a special addon for rustdesk (besides a buch of others). works great

**Comment by admiraldaro** [comment]
For anyone still following this discussion: I have released a small project implementing one of the options proposed here — synchronizing a RustDesk address book with a server of your choice.

https://github.com/admiraldaro/rust-book

Rust-Book is an unofficial open-source account and legacy address-book API server. It runs alongside the standard OSS `hbbs` and `hbbr` services rather than replacing or modifying them. You point the **API Server** setting in the regular unmodified RustDesk client to your Rust-Book URL, log in, and the client can synchronize one legacy address book associated with that account.

The project uses PHP and SQLite, includes a lightweight browser-based administration panel, and is released under the MIT license. No modified RustDesk client or server binaries are included.

To be clear about its scope, this is not a replacement for Server Pro. It currently does not provide managed devices, settings synchronization, ACLs or the newer multi-address-book API. It specifically covers account login and the legacy single-address-book flow. The current release is also still early and has so far been verified with the official Windows client 1.4.9; other client versions and platforms may behave differently.

I created it for people who are happy with the regular OSS RustDesk server but would like a small self-hosted address-book backend without having to modify or rebuild the client. Bug reports, compatibility results and contributions are welcome.
